### PR TITLE
Enable execution of sanitizers in CI for ARM64

### PR DESCRIPTION
**Pull request by jbajic**
### Scope & Purpose

Found issues:
- [x] Always build with GCC implementation of OpenMP since the clang one has issues with failing assertions (Resolved in this PR)
- [x] The found TSAN issues are in RocksDB race on global variable fixed here: https://github.com/arangodb/rocksdb/pull/11
- [x] ALUBSAN issue found during linking, which is related to using `-fsanitize=undefined` flag and int128 from abseil (https://bugs.llvm.org/show_bug.cgi?id=16404 and https://github.com/abseil/abseil-cpp/issues/841)
- [x] Upgrade snappy, undefined error(https://github.com/google/snappy/pull/148) fixed by upgrading snappy https://github.com/arangodb/arangodb/pull/22078
- [x] Fix an undefined behavior on ARM (Fixed in this PR)

Detected undefined behavior:
```
SUT was aborted: 
Report of 'arangod' in /tmp/ArangoDB/0/arangosh_6G8Wvx/shell_api_multi/coordinator_0/alubsan.log.arangod.492 contains: 
/root/project/3rdParty/velocypack/src/asm-functions.cpp:52:28: runtime error: addition of unsigned offset to 0x51800003cd1c overflowed to 0x51800003cd09
    #0 0xaafbe39858c8 in arangodb::velocypack::(anonymous namespace)::JSONStringCopyC(unsigned char*, unsigned char const*, unsigned long) /root/project/3rdParty/velocypack/src/asm-functions.cpp:52
    #1 0xaafbe395c960 in arangodb::velocypack::Parser::parseString() /root/project/3rdParty/velocypack/src/Parser.cpp
    #2 0xaafbe3961538 in arangodb::velocypack::Parser::parseArray() /root/project/3rdParty/velocypack/src/Parser.cpp:464
    #3 0xaafbe395a0cc in arangodb::velocypack::Parser::parseInternal(bool) /root/project/3rdParty/velocypack/src/Parser.cpp:68
    #4 0xaafbd9a5fb78 in parse /root/project/3rdParty/velocypack/include/velocypack/Parser.h:169
    #5 0xaafbd9a5fb78 in arangodb::RestImportHandler::parseVelocyPackLine(arangodb::velocypack::Builder&, char const*, char const*, bool&) /root/project/arangod/RestHandler/RestImportHandler.cpp:1009
    #6 0xaafbd9a5a998 in arangodb::RestImportHandler::createFromKeyValueList() /root/project/arangod/RestHandler/RestImportHandler.cpp:726
    #7 0xaafbd9a51a58 in arangodb::RestImportHandler::executeAsync() /root/project/arangod/RestHandler/RestImportHandler.cpp:111
    #8 0xaafbd981e860 in arangodb::rest::RestHandler::executeEngine() /root/project/arangod/GeneralServer/RestHandler.cpp:487
    #9 0xaafbd981d184 in arangodb::rest::RestHandler::runHandlerStateMachine() /root/project/arangod/GeneralServer/RestHandler.cpp:432
    #10 0xaafbd982178c in arangodb::rest::RestHandler::runHandler(std::function<void (arangodb::rest::RestHandler*)>) /root/project/arangod/GeneralServer/RestHandler.cpp:702
    #11 0xaafbd9792944 in operator() /root/project/arangod/GeneralServer/CommTask.cpp:697
    #12 0xaafbd9792944 in arangodb::Scheduler::WorkItem<arangodb::rest::CommTask::handleRequestSync(std::shared_ptr<arangodb::rest::RestHandler>)::$_0>::invoke() /root/project/arangod/Scheduler/Scheduler.h:200
    #13 0xaafbddd1b4bc in arangodb::SupervisedScheduler::runWorker() /root/project/arangod/Scheduler/SupervisedScheduler.cpp:479
    #14 0xaafbe32e3698 in arangodb::Thread::runMe() /root/project/lib/Basics/Thread.cpp:344
    #15 0xaafbe32e30c0 in arangodb::Thread::startThread(void*) /root/project/lib/Basics/Thread.cpp:147
    #16 0xaafbe331ad8c in ThreadStarter(void*) /root/project/lib/Basics/threads-posix.cpp:78
    #17 0xaafbd90ab224 in asan_thread_start(void*) crtbegin.c
    #18 0xff99ab6e5958  (/lib/aarch64-linux-gnu/libc.so.6+0x85958) (BuildId: d6c205bda1b6e91815f8fef45bdf56bc2239c37e)
    #19 0xff99ab74bb08  (/lib/aarch64-linux-gnu/libc.so.6+0xebb08) (BuildId: d6c205bda1b6e91815f8fef45bdf56bc2239c37e)

SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior /root/project/3rdParty/velocypack/src/asm-functions.cpp:52:28 
```

Fixes some TSAN and ALUBSAN issues on arm64 and enables sanitizers execution on CI. 

- [x] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.12.0: *(Please link PR)*
  - [ ] Backport for 3.11: *(Please link PR)*
  - [ ] Backport for 3.10: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [x] GitHub issue / Jira ticket: https://arangodb.atlassian.net/browse/COR-6
- [ ] Design document: 

---

> [!NOTE]
> Enables ARM64 sanitizer runs in CI, switches OpenMP linking to GCC’s libgomp with dynamic discovery, adds lld via presets, improves JSON import edge-case handling, and slightly increases a test timeout.
> 
> - **CI**:
>   - Enable `aarch64` enterprise workflows regardless of `sanitizer` in `.circleci/generate_config.py`.
>   - Move linker config to presets and remove explicit `-fuse-ld=lld` from `.circleci/base_config.yml` configure step.
> - **Build/CMake**:
>   - Prefer GCC OpenMP: define `ARANGODB_OPENMP_VERSION="5.1"`, force `-fopenmp=libgomp`, and dynamically locate `libgomp` for non-static builds in `CMakeLists.txt`.
>   - Add `-fuse-ld=lld` to `CMakePresets.json` (`pr-base`, `alubsan`).
> - **Server**:
>   - Improve robustness in `arangod/RestHandler/RestImportHandler.cpp`:
>     - Additional `TRI_ASSERT` bounds checks during line parsing.
>     - Treat empty/whitespace-only lines correctly; early error on missing first-line keys.
> - **Tests**:
>   - Increase BG shell wait timeout in `tests/js/client/shell/shell-index-link-parallel-drop-create.js`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 2d5c6eeccf8bf2a1af26f141b2a93b737f7b9148. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: CMakePresets.json [hunk 1]
```diff
@@ -93,6 +93,7 @@
         "STATIC_EXECUTABLES": "Off",
         "CMAKE_CXX_FLAGS": "-fsanitize=address -fsanitize=leak -fsanitize=undefined -fsanitize=float-divide-by-zero -fsanitize-address-use-after-return=never -fno-sanitize=vptr -fno-sanitize=alignment",
         "CMAKE_C_FLAGS": "-fsanitize=address -fsanitize=leak -fsanitize=undefined -fsanitize=float-divide-by-zero -fsanitize-address-use-after-return=never -fno-sanitize=alignment",
+        "CMAKE_EXE_LINKER_FLAGS": "-Wl,--build-id=sha1 -rtlib=compiler-rt -fuse-ld=lld",
```
Comment (jvolmer) [inline CMakePresets.json:96]: Out of interest: why is lld not used for the tsan build and therefore not added to the tsan configuration above?

File: arangod/RestHandler/RestImportHandler.cpp [hunk 2]
```diff
@@ -35,6 +35,7 @@
 #include "Utils/SingleCollectionTransaction.h"
 #include "VocBase/LogicalCollection.h"
 #include "VocBase/vocbase.h"
+#include "Logger/LogMacros.h"
```
Comment (jvolmer) [inline arangod/RestHandler/RestImportHandler.cpp]: nit: you can probably get rid of this import

File: CMakePresets.json (hunk 1 above)
Comment (jbajic) [reply CMakePresets.json:96]: It is used. I've just moved `CMAKE_EXE_LINKER_FLAGS` from CI to CMakePresets. So every preset inheriting from pr-base should have this set. You can check out the tsan and alubsan executions on CI in the `Configure` step and see that we do set `lld`:
- tsan: https://app.circleci.com/pipelines/github/arangodb/arangodb/39852/workflows/1af739b5-b34a-425a-9b30-1795cffdb56f/jobs/3727418
- alubsan: https://app.circleci.com/pipelines/github/arangodb/arangodb/39853/workflows/da30dfc3-64ae-4acc-a4bc-1bfd1776e8b3/jobs/3727540
